docs/guide/models/execution.rst

```diff
@@ -0,0 +1,29 @@
+Protocols and the Execution Model
+=================================
+
+Protocols in OpenFE are built on a flexible execution model. 
+Result objects are shaped by this model, and therefore some basic
+background on it can be useful when looking into the details of simulation
+results, even if most users don't need to work with the details of the
+execution model.
+
+Each protocol involves a number of steps (called ``ProtocolUnit``\ s) which occur in
+some order. Formally, this is described by a directed acyclic graph (DAG),
+so the collection of steps to run is called a ``ProtocolDAG``. A
+:class:`.Protocol` creates the ``ProtocolDAG``, and a single ``ProtocolDAG``
+should give information necessary to obtain an estimate (perhaps a very poor
+estimate) of the :math:`\Delta G`. Over the course of a campaign, a single
+:class:`.Protocol` may create multiple ``ProtocolDAG``\ s, e.g., to extend a
+simulation. NB: While independent runs can be created as separate
+``ProtocolDAG``\ s, the recommend way to do independent runs is as a
+``repeats`` part of the settings for the protocol, which puts the
+independent runs in a single ``ProtocolDAG``.
```

Comment:
This is our approach but the alchemiscale / NEQ one doesn't follow this right?

Reply:
I have no idea. This is seriously a major concern I have with the model we're using here. I think it is unintuitive, so we kind of have to shout it at the top of our lungs every time it comes up. But then if it isn't consistent... 🤷‍♂️

Reply:
🤔 yeah tbh I'm mostly concerned about what we're going to do when we start having a very of that NEQ protocol (will we just fix things ourselves and then run two different version of the Protocol or just use the current structure as-is withou tthe OEChem dependencies?). I'd be tempted to just not try to think about it, make this a todo and revisit later tbh.